ghcide/src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -565,9 +565,11 @@ getCompletions ideOpts CC { allModNamesAsNS, unqualCompls, qualCompls, importabl
         = filtPragmaCompls (pragmaSuffix fullLine)
         | otherwise
         = let uniqueFiltCompls = nubOrdOn insertText filtCompls
-          in filtModNameCompls ++ map (toggleSnippets caps withSnippets
-                                         . mkCompl ideOpts . stripAutoGenerated) uniqueFiltCompls
-                               ++ filtKeywordCompls
+          in filtModNameCompls
+          ++ filtKeywordCompls
```

Comment:
I guess you are deliberately putting the keywords before the identifiers? Makes sense to me - but is this ordering  justified for some reason?

Reply:
It's a cheap heuristic to put the most useful completions first. Keywords just seem more relevant

Reply:
Makes sense. Do we filter keywords, e.g. not allow type unless we're at the top of a statement, not allow in unless we've previously seen a let etc? (Not for this PR of course, but just thinking out loud)